lib/publiccloud/azure.pm

```diff
@@ -281,21 +303,29 @@ sub upload_img {
     $target_regions =~ s/,/ /g;    # CLI expects spaces as separation, not commas
     my $subscription = $self->provider_client->subscription;
     my $sa_url = "/subscriptions/$subscription/resourceGroups/imageGroups/providers/Microsoft.Storage/storageAccounts/$storage_account";
-    my $version = calc_img_version();
-
-    my $resource_group = $self->resource_group;
-    my $gallery = $self->image_gallery;
+    my $version = generate_img_version();
 
     ## Create image definition. This image definition can then be used by addressing it with it's
     ## /subscription/.../resourceGroups/openqa-upload/providers/Microsoft.Compute/galleries/...
     ## link.
     ## 1. Ensure the image definition in the Azure Compute Gallery exists
     ## 2. Create a new image version for that definition. Use the link to the uploaded blob to create this version
 
-    # Print image definitions as a help to debug possible conflicting definitions
+    my $resource_group = $self->resource_group;
+    my $gallery = $self->image_gallery;
```

Comment:
```suggestion
    my $gallery = $self->image_gallery();
```

Reply:
But `$self->image_gallery` is not subroutine.

See how it's defined:
```
lib/publiccloud/azure.pm: 23:has image_gallery => 'test_image_gallery';
lib/publiccloud/azure.pm:21:has resource_group => 'openqa-upload';
```